core/src/com/unciv/logic/battle/BattleDamage.kt

```diff
@@ -76,7 +76,20 @@ object BattleDamage {
             if (nearbyCivUnits.any { it.hasUnique("Bonus for units in 2 tile radius 15%") }) {
                 val greatGeneralModifier =
                     if (combatant.unit.civInfo.hasUnique("Great General provides double combat bonus")) 30 else 15
+
                 modifiers["Great General"] = greatGeneralModifier
+
+                // Hakkapeliitta unique
+                for (unique in combatant.unit.getMatchingUniques("[]% Strength when stacked with []")) {
+                    var stackedUnitsBonus = 0
+                    if (unique.params[1] == "Great General" && combatant.unit.getTile().getUnits().any { it.getMatchingUniques("Great Person - []").any { it.params[0] == "War" } })
```

Comment:
If the recognition is by name, then the unit should be by name as well.
`combatant.unit.getTile().getUnits().any{it!=combatant.unit && it.name==unique.params[1]}

Reply:
Actually matchesFilter as used below is even better.